### ISSUE TITLE
Optimization (uglify-es) breaks codemirror linting

### ISSUE THREAD

**Issue by jmesa-sistel** [angular/angular-cli#12287]
### Bug Report or Feature Request (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [X] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

npm --version
5.6.0

ng --version
Angular CLI: 6.2.2
Node: 8.11.4
OS: win32 x64
Angular: 6.1.7
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router, service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.8.2
@angular-devkit/build-angular     0.8.2
@angular-devkit/build-optimizer   0.8.2
@angular-devkit/build-webpack     0.8.2
@angular-devkit/core              0.8.2
@angular-devkit/schematics        0.8.2
@angular/cli                      6.2.2
@angular/pwa                      0.8.2
@ngtools/webpack                  6.2.2
@schematics/angular               0.8.2
@schematics/update                0.8.2
rxjs                              6.3.2
typescript                        2.9.2
webpack                           4.19.0

### Repro steps
Not easy to reproduce.
The problem is when optimization is true, then uglify-es breaks codemirror linting.
The solution is apply this commit [907fedc](https://github.com/angular/angular-cli/commit/907fedc0b8bebc227e7f70b3740babcaa4678d4a) from master to 6.x.x branch.
Or update uglifyjs-webpack-plugin to 2.0.0

### The log given by the failure

from console log:
main.d54661d3d921c6ccca26.js:1 ERROR ReferenceError: b is not defined
    at s (4.f196ba946a81605dfd12.js:1)
    at 4.f196ba946a81605dfd12.js:1
    at Wi (4.f196ba946a81605dfd12.js:1)
    at yl.operation (4.f196ba946a81605dfd12.js:1)
    at a (4.f196ba946a81605dfd12.js:1)
    at yl.performLint (4.f196ba946a81605dfd12.js:1)
    at t._next (4.f196ba946a81605dfd12.js:1)
    at t.__tryOrUnsub (main.d54661d3d921c6ccca26.js:1)
    at t.next (main.d54661d3d921c6ccca26.js:1)
    at t._next (main.d54661d3d921c6ccca26.js:1)

### Mention any other details that might be useful

Moving to terser-webpack-plugin or update uglifyjs-webpack-plugin to 2.0.0 (they have move to uglify-js in 2.0.0) fix the issue

**Comment by alan-agius4** [comment]
I am going to close this, as the issue has been already solved in `CLI 7` and there is no plan to implement `terser` on the `6.2.x` branch.

**Comment by rodries** [comment]
Then are you going to update uglifyjs-webpack-plugin to 2.0.0 in 6.2.x branch ?

**Comment by rodries** [comment]
@alan-agius4
@hansl 
Why are anybody not replying to my question?. To fix 6.2.x branch you only need to update uglifyjs-webpack-plugin to 2.0.0. Any problem with this fix?
